store/tikv/2pc.go

```diff
@@ -214,14 +230,16 @@ func (c *twoPhaseCommitter) doActionOnKeys(bo *Backoffer, action twoPhaseCommitA
 		sizeFunc = c.keyValueSize
 	}
 	// Make sure the group that contains primary key goes first.
-	batches = appendBatchBySize(batches, firstRegion, groups[firstRegion], sizeFunc, txnCommitBatchSize)
-	delete(groups, firstRegion)
+	batches = appendBatchBySize(batches, primaryRegion, groups[primaryRegion], sizeFunc, txnCommitBatchSize)
+	if len(batches) == 1 {
```

Comment:
I really don't understand in which case that `len(batches)` will be 0?

Reply:
I got the case. It is caused by a region error when committing a none-primary batch which would be re-splitted to commit.